api/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -18,129 +18,150 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
 // IngressNodeFirewallICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols
 type IngressNodeFirewallICMPRule struct {
-	// ICMPType define ICMP Type Numbers (RFC 792).
-	// +kubebuilder:validation:Required
+	// imcpType define ICMP Type Numbers (RFC 792).
+	// if its set the type field in the ICMP /ICMP6 header will match against this configured value otherwise its ignored.
+	// +kubebuilder:validation:Optional
 	// +kubebuilder:validation:Maximum:=255
 	// +kubebuilder:validation:Minimum:=0
-	// +optional
 	ICMPType uint8 `json:"icmpType,omitempty"`
 
-	// ICMPCode define ICMP Code ID (RFC 792).
-	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:Maximum:=16
+	// icmpCode define ICMP Code ID (RFC 792).
+	// if its set the code field in the ICMP /ICMP6 header will match against this configured value otherwise its ignored.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Maximum:=255
 	// +kubebuilder:validation:Minimum:=0
-	// +optional
 	ICMPCode uint8 `json:"icmpCode,omitempty"`
```

Comment:
same here, how do we distinguish between code 0 and unset? otherwise, we might want to change the description's wording and just say it defaults to 0.

Reply:
@msherif1234 How was this resolved?